scripts/src/javatests/org/oppia/android/scripts/ci/ComputeChangedFilesTest.kt

```diff
@@ -0,0 +1,826 @@
+    assertThat(exception).hasMessageThat().contains("System.exit()")
+    assertThat(pendingOutputStream.toString()).contains("Usage:")
+  }
+
+  @Test
+  fun testUtility_twoArguments_printsUsageStringAndExits() {
+    val exception = assertThrows<SecurityException>() { main(arrayOf("first", "second")) }
+
+    // Bazel catches the System.exit() call and throws a SecurityException. This is a bit hacky way
+    // to verify that System.exit() is called, but it's helpful.
+    assertThat(exception).hasMessageThat().contains("System.exit()")
+    assertThat(pendingOutputStream.toString()).contains("Usage:")
+  }
+
+  @Test
+  fun testUtility_threeArguments_printsUsageStringAndExits() {
+    val exception = assertThrows<SecurityException>() { main(arrayOf("first", "second", "three")) }
+
+    // Bazel catches the System.exit() call and throws a SecurityException. This is a bit hacky way
+    // to verify that System.exit() is called, but it's helpful.
+    assertThat(exception).hasMessageThat().contains("System.exit()")
+    assertThat(pendingOutputStream.toString()).contains("Usage:")
+  }
+
+  @Test
+  fun testUtility_directoryRootDoesNotExist_throwsException() {
+    val exception = assertThrows<IllegalStateException>() {
+      main(arrayOf("fake", "alsofake", "andstillfake", "compute_all_files=false"))
+    }
+
+    assertThat(exception).hasMessageThat().contains("Expected 'fake' to be a directory")
+  }
+
+  @Test
+  fun testUtility_invalid_lastArgument_throwsException() {
+    val exception = assertThrows<IllegalStateException>() {
+      main(arrayOf("fake", "alsofake", "andstillfake", "compute_all_filess=false"))
+    }
+
+    assertThat(exception).hasMessageThat()
+      .contains("Expected last argument to start with 'compute_all_files='")
+  }
+
+  @Test
+  fun testUtility_invalid_lastArgumentValue_throwsException() {
+    val exception = assertThrows<IllegalStateException>() {
+      main(arrayOf("fake", "alsofake", "andstillfake", "compute_all_files=blah"))
+    }
+
+    assertThat(exception).hasMessageThat()
+      .contains("Expected last argument to have 'true' or 'false' passed to it, not: 'blah'")
+  }
+
+  @Test
+  fun testUtility_emptyDirectory_throwsException() {
+    val exception = assertThrows<IllegalStateException>() { runScript(currentHeadHash = "ad") }
+
+    assertThat(exception).hasMessageThat().contains("run from the workspace's root directory")
+  }
+
+  @Test
+  fun testUtility_emptyWorkspace_returnsNoTargets() {
+    // Need to be on a feature branch since the develop branch expects there to be files.
+    initializeEmptyGitRepository()
+    switchToFeatureBranch()
+    createEmptyWorkspace()
+
+    val reportedFiles = runScript()
+
+    // An empty workspace should yield no files.
+    assertThat(reportedFiles).isEmpty()
+  }
+
+  @Test
+  fun testUtility_developBranch_returnsAllFiles() {
+    initializeEmptyGitRepository()
+    switchToFeatureBranch()
```

Comment:
Aren't this & the other 'develop' tests actually running on a feature branch? Is the script actually behaving correctly?

Reply:
Apologies, mistakenly switched to feature branch with the test `testUtility_developBranch_returnsAllFiles` fixed that now.
And can I know if any other develop test are running in the feature branch, I can't find anything else (am I missing anything).

Reg the behaviour: It retrieves to check if all the committed files are computed and that is the intended result we are looking for. If I'm not wrong? can I get more context if that was not the expected script behaviour.

Reply:
If the question is on **"why the even after switching to feature branch it returned the list of files?"** (making the test case PASS) - (in case) to address that, -> I had committed the files only after switching to the feature branch, so it collected all the files in the feature branch and computed it.

#

If we had tried to commit the files before switching it would have thrown an error -

![image](https://github.com/user-attachments/assets/31fc51d9-60f3-4cd9-86aa-07aa5234b8c2)

The updated test now handles it correctly in the develop branch itself.

---

But the comment mentioned here is yet to be addressed - it makes sense, https://github.com/oppia/oppia-android/pull/5465/files#r1710639970 and I will probably need to remove untracked files from the changed files list.

Reply:
I think the committed files probably makes sense as the source of truth. Is there anything else that needs to be addressed here? It seems you fixed the discrepancies in the latest changes unless I missed something.

Reply:
>Is there anything else that needs to be addressed here?

Sorry, forgot to update it here.

>But the comment mentioned here is yet to be addressed - it makes sense, https://github.com/oppia/oppia-android/pull/5465/files#r1710639970 and I will probably need to remove untracked files from the changed files list.

The above mentioned changed was implemented by only considering committed files. Now the scripts function as expected.

Reply:
Ah excellent--it seemed like the changes were completed but I wanted to make sure. Thanks!